From 929872513cf14c4c11df1b144b97595301f69475 Mon Sep 17 00:00:00 2001
From: Guillem Jover <gjover@sipwise.com>
Date: Fri, 25 Aug 2017 14:12:15 +0200
Subject: [PATCH] apps/diameter_client: Do not access the OpenSSL BIO struct
 directly

Use a proper accessor instead of messing with the struc directly
which has become opaque in latest OpenSSL versions.
---
 apps/diameter_client/lib_dbase/tcp_comm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/apps/diameter_client/lib_dbase/tcp_comm.c b/apps/diameter_client/lib_dbase/tcp_comm.c
index d9fea4591..1290fc3a3 100644
--- a/apps/diameter_client/lib_dbase/tcp_comm.c
+++ b/apps/diameter_client/lib_dbase/tcp_comm.c
@@ -74,7 +74,7 @@ long tcp_ssl_dbg_cb(BIO *bio, int oper, const char *argp,
   switch (oper) {
   case BIO_CB_WRITE: {
     char buf[256];
-    snprintf(buf, 256, "%s: %s", argp, bio->method->name);
+    snprintf(buf, 256, "%s: %s", argp, BIO_method_name(bio));
     INFO("%s", buf);
   } break;
 
